data/debug-cli/docker_debug.yaml

```diff
@@ -0,0 +1,166 @@
+  - builtins                       Show builtin commands.
+  
+  Checks:
+  ✓ distro:            Debian GNU/Linux 12 (bookworm)
+  ✓ entrypoint linter: no errors (run 'entrypoint' for details)
+
+  Note: This is a sandbox shell. All changes will not affect the actual image.
+                                                                                                                     Version: 0.0.18 (BETA)
+  root@a8758716bb6a / [nginx:latest]
+  docker >
+  ```
+  
+  The console now has a shell in the read-only filesystem of the nginx image.
+  You can now run `entrypoint` to inspect the entrypoint or run `builtins` to
+  see all builtin commands.
+  
+  ### Example 2: Dumping a file from an image to stdout by using the -c mode
+  
+  Use the `-c` (`--command`) flag to directly evaluate a command instead of
+  running the shell in interactive mode.
+  
+  The following example dumps the default `index.html` file of nginx to stdout.
+
+  ```console
+  $ docker debug nginx -c "cat /usr/share/nginx/html/index.html"
+  <!DOCTYPE html>
+  <html>
+  <head>
+  <title>Welcome to nginx!</title>
+  <style>
+  html { color-scheme: light dark; }
+  body { width: 35em; margin: 0 auto;
+  font-family: Tahoma, Verdana, Arial, sans-serif; }
+  </style>
+  </head>
+  <body>
+  <h1>Welcome to nginx!</h1>
+  <p>If you see this page, the nginx web server is successfully installed and
+  working. Further configuration is required.</p>
+  
+  <p>For online documentation and support please refer to
+  <a href="http://nginx.org/">nginx.org</a>.<br/>
+  Commercial support is available at
+  <a href="http://nginx.com/">nginx.com</a>.</p>
+  
+  <p><em>Thank you for using nginx.</em></p>
+  </body>
+  </html>
+  ```
+  
+  ### Example 3: Get a shell into a stopped container via SSH
+  
+  Use the `--host` flag to connect to a different Docker Engine other than the
+  default.
+  
+  The following example connects to a remote Docker instance via SSH.
+  
+  ```console
+  $ docker debug --host ssh://root@example.org my-container
+  ```
+  
+  The `--host` flag may be used with all other examples, including getting a
+  shell into an image.
+  
+  ### Example 4: Get a shell into a running container that has no shell
+  
+  ```console
+  $ docker run --detach --rm --volume=/var/run/docker.sock:/var/run/docker.sock -p 8080:8080 --name my-container amir20/dozzle
+  $ docker debug my-container
+  ```
+  
+  Once the shell is attached to the container, you can add further tools with
+  the built-in `install` command. For example, run `install nmap`. Adding
+  further tools to your debugging shell does not modify the target container. 
+  
+  The built-in `forward` command lets you use remote tools. The following example starts a web file browser.
+  
+  ```console
+  docker > forward --port 9000:9012 -- webfsd -F -p 9012
+  ```
+  
+  You can access the file browser on your host's browser at
+  http://localhost:9000.
```

Comment:
I am not sure I understand this example. Where is `webfsd` expected to be installed? On the host?

Reply:
So maybe we should group the examples into 
* category A: the use cases where `docker debug` can be applied
* category B: extra things that are possible in the debug shell / special tools we added to your toolbox

Currently we have category A and two cases of category B are mixed within (the `entrypoint` and the `forward` example). Probably we should move them out and create a section `Things you can do inside the debug shell that you would not expect` (better title welcome). And then there we'd have
1. installing/removing tools
2. checking the entry point
3. dynamic port forwarding using the `forward` command
4. ...(there will come more in future)

Reply:
Removed this example for this release as it seems forward may be buggy and it isn't listed as part of the builtins for this reason.

Reorganized the examples by options, builtin commands, and other. I think any quirky behavior or use-cases we can add to the "other" section. To highlight use-cases, we can also consider adding it to the language-specific guides, or create something like a Debug with Docker guide for more visibility.